test/persistence-hooks.suite.js

```diff
@@ -3240,13 +3249,17 @@ module.exports = function(dataSource, should, connectorCapabilities) {
           { id: existingInstance.id, name: 'updated name' },
           function(err, instance) {
             if (err) return done(err);
-            ctxRecorder.records.should.eql(aCtxForModel(TestModel, {
+            var expectedContext = aCtxForModel(TestModel, {
               data: {
                 id: existingInstance.id,
                 name: 'updated name',
               },
-              isNewInstance: false,
-            }));
+            });
+
+            if (dataSource.connector.upsertWithWhere) {
+              expectedContext.isNewInstance = false;
```

Comment:
For non-atomic implementation of `upsertWithWhere` on update, it called updateAttributes. `loaded` hook of `updateAttributes` does not provide `isNewInstance`; please see [here](https://github.com/strongloop/loopback-datasource-juggler/blob/72200ce935367cffc7c7df3b6cffc625acb8192a/lib/dao.js#L3091-L3097) although it is a simple fix, since it is out of scope of this PR and the change needs some other test fixes as well, so I would say it should be fixed in another PR. I will take care of it soon.